fractions/microprofile/microprofile-jwt/src/main/java/org/wildfly/swarm/microprofile/jwtauth/deployment/auth/JWTAuthMechanismFactory.java

```diff
@@ -81,18 +81,22 @@ public AuthenticationMechanism create(String mechanismName, FormParserFactory fo
             }
 
             String publicKeyPemEnc = getResource(properties, "signerPubKey", "MP-JWT-SIGNER");
-            if (publicKeyPemEnc == null) { // signerPubKey and MP-JWT-Signer was empty, now trying for JWKS URI.
-                String jwksUri = getResource(properties, "jwksUri", "MP-JWT-JWKS");
-                if (jwksUri != null) {
-                    contextInfo.setPublicKeyLocation(jwksUri);
-
-                    String jwksRefreshInterval = getResource(properties, "jwksRefreshInterval", "MP-JWT-JWKS-REFRESH");
-                    if (jwksRefreshInterval == null) {
-                        throw new IllegalStateException("JWKS Refresh Interval should be set when JWKS URI is used.");
+            if (publicKeyPemEnc == null) { // MP-JWT-Signer was empty, now trying for the key location.
+                String publicKeyLocation = getResource(properties, "signerPubKeyLocation", "MP-JWT-SIGNER-KEY-LOCATION");
+                if (publicKeyLocation == null) {
+                    publicKeyLocation = getResource(properties, "jwksUri", "MP-JWT-SIGNER-KEY-LOCATION");
+                }
+                if (publicKeyLocation != null) {
+                    contextInfo.setPublicKeyLocation(publicKeyLocation);
+                    if (publicKeyLocation.startsWith("https:")) {
+                        String jwksRefreshInterval = getResource(properties, "jwksRefreshInterval", "MP-JWT-JWKS-REFRESH");
+                        if (jwksRefreshInterval == null) {
+                            throw new IllegalStateException("JWKS Refresh Interval should be set when JWKS URI is used.");
```

Comment:
Should we say "must be set" instead of "should"?

Reply:
I'm not even sure this null check is needed, the code was there before so I just kept, there is a default value of 60 mins.

Reply:
I'm not sure if we should throw an exception if there's a default, but well...

Reply:
I'll just update it to set the context only if it is not null, Jose4j code has also some default value